src/toolkits/feature_engineering/word_trimmer.cpp

```diff
@@ -148,7 +144,7 @@ void word_trimmer_topk_index_mapping(const gl_sarray& src,
     size_t end_idx = src_size * (thread_idx + 1) / num_threads;
 
     for (const auto& v: src.range_iterator(start_idx, end_idx)) {
-      switch(run_mode) {
+      switch(v.get_type()) {
```

Comment:
Now that we check the type of each element individually, do we have to check that every element in `src` has the same type?

Reply:
you can think undefined here means the value is missing, not saying the **type** is undefined. You can rephrase it as this element should be type `blah` but the value is missing.